backend/graphql/resolvers/courseResolvers.ts

```diff
@@ -16,14 +21,70 @@ import UserService from "../../services/implementations/userService";
 import IEmailService from "../../services/interfaces/emailService";
 import IUserService from "../../services/interfaces/userService";
 import { Role } from "../../types";
-import { CourseVisibilityAttributes } from "../../models/course.model";
+import { CourseVisibilityAttributes, Module } from "../../models/course.model";
 import { assertNever } from "../../utilities/errorUtils";
 
 const courseService: ICourseService = new CourseService();
 const userService: IUserService = new UserService();
 const emailService: IEmailService = new EmailService(nodemailerConfig);
 const authService: IAuthService = new AuthService(userService, emailService);
 
+function serializeToModuleDTO(modules: { [id: string]: Module }): ModuleDTO[] {
+  return Object.entries(modules).map(([key, val]) => ({ id: key, ...val }));
+}
+
+function deserializeModuleDTO(
+  moduleDTOs: ModuleDTO[],
+): { [id: string]: Module } {
```

Comment:
Think we should make `{ [id: string]: Module }` its own type, it's used a decent amount and is something that should be very clear

Reply:
Yeahh I should've done that, just got lazy, I'll make it its own type